airflow/providers/airbyte/operators/airbyte.py

```
@@ -0,0 +1,77 @@
+from airflow.utils.decorators import apply_defaults
+
+
+class AirbyteTriggerSyncOperator(BaseOperator):
+    """
+    This operator allows you to submit a job to an Airbyte server to run a integration
+    process between your source and destination.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:AirbyteTriggerSyncOperator`
+
+    :param airbyte_conn_id: Required. The name of the Airflow connection to get connection
+        information for Airbyte.
+    :type airbyte_conn_id: str
+    :param connection_id: Required. The Airbyte ConnectionId UUID between a source and destination.
+    :type connection_id: str
+    :param asynchronous: Optional. Flag to get job_id after submitting the job to the Airbyte API.
+    :type asynchronous: bool
+    :param api_version: Optional. Airbyte API version.
+    :type api_version: str
+    :param timeout: Optional. The amount of time, in seconds, to wait for the request to complete.
+    :type timeout: float
+    """
+
+    template_fields = ('connection_id',)
+
+    @apply_defaults
+    def __init__(
+        self,
+        connection_id: str,
+        airbyte_conn_id: str = "default_airbyte_conn",
+        asynchronous: Optional[bool] = False,
+        api_version: Optional[str] = "v1",
+        timeout: Optional[float] = 3600,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.airbyte_conn_id = airbyte_conn_id
+        self.connection_id = connection_id
+        self.timeout = timeout
+        self.api_version = api_version
+        self.asynchronous = asynchronous
+
+    def execute(self, context) -> None:
+        """Create Airbyte Job and wait to finish"""
+        hook = AirbyteHook(airbyte_conn_id=self.airbyte_conn_id, api_version=self.api_version)
+        job_object = hook.submit_sync_connection(connection_id=self.connection_id)
+        job_id = job_object.json().get('job').get('id')
```

``job_object.json().get('job')`` can return `None` and we will have a AttributeError. We should use [] to access dict elements or add more checks for None values.

---

@mik-laj using the `[]` should I use a try/except block to manage the error or can let it throw like this?

---

@marcosmarxm  i think you only need to catch an error if you will change the behavior depending on the error.

in general simply catching and re-raising is not helpful, except in rare cases where the error message is misleading or completely useless and you need to produce a more helpful message.